flink/v1.17/flink/src/main/java/org/apache/iceberg/flink/sink/shuffle/DataStatisticsOperator.java

```diff
@@ -40,50 +40,49 @@
  * shuffle record to improve data clustering while maintaining relative balanced traffic
  * distribution to downstream subtasks.
  */
-class DataStatisticsOperator<T, K> extends AbstractStreamOperator<DataStatisticsOrRecord<T, K>>
-    implements OneInputStreamOperator<T, DataStatisticsOrRecord<T, K>>, OperatorEventHandler {
+class DataStatisticsOperator<D extends DataStatistics<D, S>, S>
```

Comment:
Ok, I think this DataStatisticsOperator is good for collecting some statistical information. It seems that we also need an Operator to determine the partitionID, such as PartitionIdAssignerOperator and then pass ```org.apache.flink.api.java.functions.IdPartitioner``` From custom data distribution, I haven’t seen the design of this piece in the design document, can you briefly introduce the follow-up implementation

Reply:
Yes, this is only for collecting statistics to guide the partitioner decision. we will implement a custom range partitioner (Flink )  that splits the value into ranges (one for each assigner writer subtask) based on the statistics.

Reply:
According to my understanding, each writer will be responsible for one or more partitions, and we will distribute the data arriving from upstream to the corresponding writers, right?

Reply:
@hililiwei that is correct. custom range partitioner for Flink DataStream will distribute the data to writer subtasks with good clustering based on the data statistics.